src/buffer/out/textBuffer.cpp

```diff
@@ -1528,9 +1530,19 @@ const til::point TextBuffer::GetGlyphEnd(const til::point pos, bool accessibilit
 bool TextBuffer::MoveToNextGlyph(til::point& pos, bool allowExclusiveEnd, std::optional<til::point> limitOptional) const
 {
     const auto bufferSize = GetSize();
-    const auto limit{ limitOptional.value_or(til::point{ bufferSize.EndExclusive() }) };
+    bool pastEndInclusive;
+    til::point limit;
+    {
+        // if the limit is past the end of the buffer,
+        // 1) clamp limit to end of buffer
+        // 2) set pastEndInclusive
+        const til::point endInclusive{ bufferSize.RightInclusive(), bufferSize.BottomInclusive() };
+        const auto val = limitOptional.value_or(endInclusive);

```

Comment:
The old code uses `bufferSize.EndExclusive()` and not the "inclusive" variant?

Reply:
The only reason we used `EndExclusive` before was for the `CompareInBounds` call below. So instead, we're using `EndInclusive` (a valid point in the buffer), so that `CompareInBounds` works fine. Then we add 1 if we were past `EndInclusive` (i.e. `EndExclusive`).